woocommerce/woocommerce-android
# issue/6111-step1-site-picker-changes

Updates the FluxC changes for #6111 .

### Description
This PR update the site settings and API version check methods in Woo into suspendable methods and removes the usage of [deprecated event bus architecture](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture).

The corresponding FluxC PRs [1](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2331) and [2](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2332) need to be reviewed and merged before this PR can be merged. 

**This is a breaking change so it would be good to merge the FluxC PR and this PR together after review**

### Testing instructions
There are no user facing changes here. Just go through some of the site picker flows in the app to make sure there are no breakages. 

- Login with `Enter your site address` flow and ensure that you are automatically connected to that store after successful login.
- Test using a Jetpack CP site to ensure that the Jetpack CP flow works correctly.
- Try switching stores after successful login and ensure that there are no changes to the flow.
- Try logging in with your email address and verify that you are able to select a store from the site picker screen.
